spec/automation/recurring_data_explorer_result_topic_spec.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe "RecurringDataExplorerResultTopic" do
+  fab!(:admin)
+
+  fab!(:user)
+  fab!(:another_user) { Fabricate(:user) }
+  fab!(:group_user) { Fabricate(:user) }
+  fab!(:not_allowed_user) { Fabricate(:user) }
+  fab!(:topic)
+
+  fab!(:group) { Fabricate(:group, users: [user, another_user]) }
+  fab!(:another_group) { Fabricate(:group, users: [group_user]) }
+
+  fab!(:automation) do
+    Fabricate(:automation, script: "recurring_data_explorer_result_topic", trigger: "recurring")
+  end
+  fab!(:query)
+  fab!(:query_group) { Fabricate(:query_group, query: query, group: group) }
+  fab!(:query_group) { Fabricate(:query_group, query: query, group: another_group) }
+
+  before do
+    SiteSetting.data_explorer_enabled = true
+    SiteSetting.discourse_automation_enabled = true
+
+    automation.upsert_field!("query_id", "choices", { value: query.id })
+    automation.upsert_field!("topic_id", "text", { value: topic.id })
+    automation.upsert_field!(
+      "query_params",
+      "key-value",
+      { value: [%w[from_days_ago 0], %w[duration_days 15]] },
+    )
+    automation.upsert_field!(
+      "recurrence",
+      "period",
+      { value: { interval: 1, frequency: "day" } },
+      target: "trigger",
+    )
+    automation.upsert_field!("start_date", "date_time", { value: 2.minutes.ago }, target: "trigger")
+  end
+
+  context "when using recurring trigger" do
+    it "sends the post at recurring date_date" do
+      freeze_time 1.day.from_now do
+        expect { Jobs::DiscourseAutomation::Tracker.new.execute }.to change {
+          topic.reload.posts.count
+        }.by(1)
+
+        expect(topic.posts.last.raw).to include("Scheduled Report for #{query.name}")
```

Comment:
since we are generating stuff here ... might as well test that the query is rendered, even something basic like query is "select 'abctestabc'" and look for the text abctestabc?

additionally we need url support for our specific use case: 

eg: select `abc,https://test.com` as some_url 

data explorer has special handling for _url.

Reply:
Yep makes sense, will do that and handle URLs :+1:

Reply:
The URL special handling is on the client so we don't need to test for that here. For our internal example, the DE table generator returns something like this:

```
| team | in_progress_url | up_next_url | maybe_next_url | maybe_later_url | unprioritized_url |
| :----- | :----- | :----- | :----- | :----- | :----- |
| blah | 4,https://test.com/xyz | 13,https://test.com/xyz | 18,https://test.com/xyz | 15,https://test.com/xyz | 11,https://test.com/xyz |
```

So the client itself must wrap the count with the URL